server/src/http/feature_refresher.rs

```diff
@@ -299,6 +311,97 @@ impl FeatureRefresher {
+                                {
+                                    debug!(
+                                        "Connected to unleash! I should populate my flag cache now.",
+                                    );
+
+                                    // very rough handling of client features.
+                                    let features: ClientFeatures =
+                                    serde_json::from_str(&event.data).unwrap();
+                                    refresher.handle_client_features_updated(TokenRefresh::new(token, None), features);
+                                }
+                                // Unleash has updated. This is where we send data to listeners.
+                                eventsource_client::SSE::Event(event)
+                                    if event.event_type == "unleash-updated" =>
+                                {
+                                    debug!(
+                                        "Got an unleash updated event. I should update my cache and notify listeners.",
+                                    );
+
+                                    // store the data locally
+                                    let features: ClientFeatures =
+                                    serde_json::from_str(&event.data).unwrap();
+                                    refresher.handle_client_features_updated(TokenRefresh::new(token, None), features);
+
+                                    // send the data to the broadcaster. This should probably just send the new
+                                    // feature set OR even just a "filter flags"
+                                    // function. The broadcaster will take care
+                                    // of filtering the flags per listener.
+                                    // let data = Data::new(event.data).event("unleash-updated");
+                                    // broadcaster.rebroadcast(actix_web_lab::sse::Event::Data(data)).await;
+                                    refresher.broadcaster.broadcast().await;
+                                }
+                                eventsource_client::SSE::Event(event) => {
+                                    debug!(
+                                        "Got an SSE event that I wasn't expecting: {:#?}",
+                                        event
+                                    );
+                                }
+                                eventsource_client::SSE::Connected(_) => {
+                                    debug!("SSE Connection established");
+                                }
+                                eventsource_client::SSE::Comment(_) => {
+                                    // purposefully left blank.
+                                },
+                            }
+                        }
+                    })
+                    .map_err(|e| warn!("Error in SSE stream: {:?}", e));
```

Comment:
handle 404s etc, maybe launch polling if can't connect

Reply:
This is a more complex topic. I suggest we move handling this one to a little later.